### PR TITLE
fixes bug where rotational moves used incorrect positions; also resolves two other issues

### PR DESCRIPTION
**Pull request by sgill2**
Addresses #25 and cleans up some related code handling error reporting.
Also addresses #26 and removes the self.nc_pos variable in the SimNCMC class.
More explicitly this fixes an issue where in a rotational move, the positions from the beginning point of the ncmc step was used instead of the current positions in the ncmc iteration.
Finally, cleans up some print statements in the code.

### CODE REVIEW

File: blues/ncmc.py [hunk 1]
```diff
@@ -173,15 +173,14 @@ def __init__(self, temperature, residueList, **kwds):
...
-        self.nc_pos = None
         self._storage = None
         self.temperature = temperature
         kB = unit.BOLTZMANN_CONSTANT_kB * unit.AVOGADRO_CONSTANT_NA
         kT = kB * temperature
         beta = 1.0 / kT
         self.beta = beta
 
-    def get_particle_masses(self, system, residueList=None):
+    def get_particle_masses(self, system, set_self=True, residueList=None):
```
Comment (davidlmobley) [inline blues/ncmc.py]: Doc string for function?

Comment (davidlmobley) [inline blues/ncmc.py]: (e.g., what does it do if this is/is not true? I should be able to use it without having to read the code...)

File: blues/ncmc.py [hunk 2]
```diff
@@ -263,6 +263,9 @@ def calculate_com(self,  pos_state, total_mass=None, mass_list=None, residueList
         lig_coord = lig_coord*unit.nanometers
         copy_coord = copy.deepcopy(lig_coord)
         #mass corrected coordinates (to find COM)
+        print('mass_list', mass_list)
```
Comment (davidlmobley) [inline blues/ncmc.py]: Why are we printing these? Is this for debug purposes? If so, can you just make an optional `debug` argument and have them printed only if debug is True? Presumably you don't want a log file somewhere filling up with tens of thousands of debug statements.

File: blues/ncmc.py [hunk 3]
```diff
@@ -345,27 +348,30 @@ def create_normalSystem(self, coord_file, top_file):
...
-   
-    def rotationalMove(self, residueList=None):
+
+    def rotationalMove(self, context=None, residueList=None,):
         """
         Function to be used in movekey. Performs a rotation around the center of mass
         of the ligand.
         Arguments
         ---------
         generally none
```
Comment (davidlmobley) [inline blues/ncmc.py]: This takes an optional `context` argument now, so it should be explained what that does. Also, what happens if I give a residue list?

File: blues/ncmc.py [hunk 4]
```diff
@@ -534,44 +533,23 @@ def runSim(self, md_simulation, nc_context, nc_integrator, dummy_simulation, mov
...
                     print('Error:', e)
                     stateinfo = md_simulation.context.getState(True, True, False, False, False, periodic)
-                    print(oldPos)
                     last_x, last_y = np.shape(oldPos)
-                    print(np.reshape(oldPos, (1, last_x, last_y)))
                     reshape = (np.reshape(oldPos, (1, last_x, last_y))).value_in_unit(unit.nanometers)
-                    print(np.shape(reshape))
-                    print(newinfo.getPotentialEnergy())
-                    print(newinfo.getKineticEnergy())
                     print(oldPE)
```
Comment (davidlmobley) [inline blues/ncmc.py]: Are these prints debug statements? Usually not very useful to user to have a print of something without hvaing it say what is being printed.

File: blues/ncmc.py [hunk 5]
```diff
@@ -534,44 +533,23 @@ def runSim(self, md_simulation, nc_context, nc_integrator, dummy_simulation, mov
...
             oldPos = md_stateinfo.getPositions(asNumpy=True)
             oldVel = md_stateinfo.getVelocities(asNumpy=True)
             nc_integrator.reset()
             nc_context.setPositions(oldPos)
-            self.nc_pos = oldPos
             nc_context.setVelocities(oldVel)
-            
+
         acceptRatio = accCounter/float(niter)
         print(acceptRatio)
```
Comment (davidlmobley) [inline blues/ncmc.py]: Again, probably want to say what is being printed.

**Comment by davidlmobley** [comment]
If I understand correctly, @sgill2 , this is the one which fixes a critical bug relating to which positions are being used? If so, this should be noted more clearly in the pull request summary.

(We're not yet in a world where outside folks are using this/watching it, but we will be there soon -- likely beginning after the first week in March -- so we need to be sure that what we put here is explained well enough that outside folks can follow what is going on.)

**Comment by sgill2** [comment]
@davidlmobley updated the pull request summary and merged with the latest master branch, which has the recently implemented tests that this PR passes.

File: blues/ncmc.py [hunk 6]
```diff
@@ -551,6 +560,6 @@ def runSim(self, md_simulation, nc_context, nc_integrator, dummy_simulation, mov
             nc_context.setVelocities(oldVel)
 
         acceptRatio = accCounter/float(niter)
-        print(acceptRatio)
+        print('acceptance ratio', acceptRatio)
```
Comment (davidlmobley) [inline blues/ncmc.py]: Still a danger that these will fill a log file?

**Comment by davidlmobley** [comment]
@sgill2 - this is good to merge once you address any of the remaining "should you really be printing that by default?" type issues.

**Comment by sgill2** [comment]
Oh yeah, I forgot to mention that my latest commit tried to make the print statements more explicit and removed really unnecessary ones.

**Comment by davidlmobley** [comment]
Yes, the latest commit at least documented what it was printing, which was much better. (You want to assume that your users are not going to read the code.) But, often, I write code so that functions will take an optional `debug` argument which, if True, prints lots of stuff, but if it is not true, it doesn't print anything except stuff that everyone will want to know. I was wondering if that should be applied to any of these.

**Comment by sgill2** [comment]
Hmm, I somewhat have that here with runSim taking an optional verbose argument, which prints other sometimes interesting things like the protocol/shadow work contributions at each NCMC step. I think it's printing only really relevant information now, but I'll keep that in mind when I'm looking over the code in the future.

**Comment by davidlmobley** [comment]
OK, sounds good!

**Pull request merged**